bridge/src/bridge/deploy.rs

```diff
@@ -116,3 +159,72 @@ impl<T: Transport + Clone> Future for Deploy<T> {
 		}
 	}
 }
+
+/// writes useful information about the deployment into `dir`.
+/// REMOVES `dir` if it already exists!
+/// helps with troubleshooting and verification (https://ropsten.etherscan.io/verifyContract)
+/// of deployments.
+/// information includes:
+/// - solc version used
+/// - git commit
+/// - contract source code
+/// - contract address
+/// - hash of transaction the contract got deployed in
+/// - contract byte code
+/// - input data for contract creation transaction
+/// - ...
+pub fn write_deployment_info<P: AsRef<Path>>(
+	dir: P,
+	contract_address: &H160,
+	transaction_hash: &H256,
+	contract_name: &str,
+	abi: &str,
+	bytecode_hex: &str,
+	deployed_bytecode_hex: &str,
+) -> Result<(), Error> {
+	let dir = dir.as_ref();
+
+	if Path::new(dir).exists() {
+		fs::remove_dir_all(dir)?;
+	}
+	fs::create_dir(dir)?;
+
+	let mut file = File::create(dir.join("bridge_version"))?;
+	file.write_all(env!("CARGO_PKG_VERSION").as_bytes())?;
+
+	let mut file = File::create(dir.join("commit_hash"))?;
+	file.write_all(env!("GIT_HASH").as_bytes())?;
+
+	let mut file = File::create(dir.join("compiler"))?;
+	file.write_all(env!("SOLC_VERSION").as_bytes())?;
+
+	let mut file = File::create(dir.join("optimization"))?;
+	file.write_all("yes".as_bytes())?;
+
+	let mut file = File::create(dir.join("contract_name"))?;
+	file.write_all(contract_name.as_bytes())?;
+
+	let mut file = File::create(dir.join("contract_address"))?;
+	file.write_all(contract_address.to_hex().as_bytes())?;
+
+	let mut file = File::create(dir.join("transaction_hash"))?;
+	file.write_all(transaction_hash.to_hex().as_bytes())?;
+
+	let mut file = File::create(dir.join("deployed_bytecode"))?;
+	file.write_all(bytecode_hex.as_bytes())?;
+
+	assert_eq!(
+		bytecode_hex,
+		&deployed_bytecode_hex[..bytecode_hex.len()],
+		"deployed byte code is contract bytecode followed by constructor args; qed"
+	);
+
+	let constructor_arguments_bytecode = &deployed_bytecode_hex[bytecode_hex.len()..];
+
+	let mut file = File::create(dir.join("constructor_arguments_bytecode"))?;
+	file.write_all(constructor_arguments_bytecode.as_bytes())?;
+
+	File::create(dir.join("abi"))?.write_all(abi.as_bytes())?;
+
+	Ok(())
```

Comment:
why are they so many files?

Reply:
i think it's the simplest way to achieve this.

we could merge smaller fields into one json or toml file but that makes it more complicated

Reply:
that's what I thought. I don't have a strong opinion about this and I was just asking to confirm that there is no other reason :)

Reply:
the nice thing about this and goal of the change is that we can now just ask someone for the `deploy-{home|foreign}-{tx-hash}` folder and have all the information we could need for troubleshooting : ) the folder also contains all information one needs to use https://etherscan.io/verifyContract